src/articles/RecommendedArticles.js

```diff
@@ -0,0 +1,51 @@
+import { useState } from "react";
+import LoadingAnimation from "../components/LoadingAnimation";
+import { setTitle } from "../assorted/setTitle";
+import strings from "../i18n/definitions";
+
+import ArticlePreview from "./ArticlePreview";
+
+import SortingButtons from "./SortingButtons";
+
+import * as s from "../components/TopMessage.sc";
+
+export default function OwnArticles({ api }) {
+  const [articleList, setArticleList] = useState(null);
+
+  let originalList = articleList;
+
+  if (articleList == null) {
```

Comment:
Ok, at a second glance, this is a bit broken, if I'm not wrong. In particular: 

- `originalList` is always going to be pointing to the current state of the `articleList` so you can never return to the original ordering with this approach. look again at the `FindArticles` component, and follow that pattern with a separate state variable for each of the two concepts

Reply:
also, the conditioning on `articleList == null` is not idiomatic, for initializing a react component. The idiomatic solution is again, like in `FindArticles`:

```
  useEffect(() => {
    api.getRecommendedArticles((articles) => {
      setArticleList(articles);
      setOriginalList(articles);
    });

    setTitle("Recommended Articles");
  }, []);

  if (articleList == null) {
    return <LoadingAnimation />;
  }

  if (articleList.length === 0) {
    return <LoadingAnimation />;
  }
```

If you then look at this, you'll see another problem: `articleList.length === 0` -- this situation is not a case for a loading animation. `list ==null` is loading. `length==0` means the api returned an empty list. For some reason there was no recommendation made. a corresponding message should then be presented to the learner.

Reply:
The code in RecommendedArticles is taken from OwnArticles, which is the one saved articles are stored in. I figured that it was quite a similar page, so I used the same implementation in the for you page.

I can implement the changes that you proposed using the pattern from FindArticles, into RecommendedArticles. But just to let you know, that this problem will still persist in OwnArticles.

Reply:
ownArticles is fixed in https://github.com/zeeguu/web/commit/d87a533c464e562b93c2a8a8a3882764e843e5e1

Reply:
@mircealungu I pushed the change with inspiration from what you did in OwnArticles. It's fetches from the api on initialization of the component. It updates the articleList and sets the originalArticles following the pattern from FindArticles and OwnArticles. Added loading animation to when the articleList is null, and added a string message when the articleList is 0, for users that might find the routing by chance.